Create FAQ.md (#1416)
zkSNACKs/WalletWasabi: WalletWasabi.Documentation/FAQ.md

# Pre-Mix

### My wallet can't send to Bech32 addresses - what wallets can I use instead? 

Wasabi generates Bech32 addresses only. These addresses start with the characters bc1... Some wallets/exchanges don't yet support this type of address and my give an error message (e.g. "unknown bitcoin address"). The solution is to manage your funds with a wallet which does support Bech32, [see list](https://en.bitcoin.it/wiki/Bech32_adoption).

Be careful, if you send all your coins from an old wallet to a new wallet (from the table above) in one transaction then you will merge all your coins which is bad for privacy - instead, **send the coins individually** or if possible **import the seed in the new wallet**.

# Mixing

### What are the fees?

You currently pay a fee of 0.003% * anonymity set. If the anonymity set of a coin is 50 then you paid 0.003% * 50 (=0.15%). If you set the target anonymity set to 53 then wasabi will continue mixing until this is reached, so you may end up with an anonymity set of say 60, and you will pay 0.003% * 60 (=0.18%).  

> There are also edge cases where you don't pay the full fee or where you pay more. For example if you're the smallest registrant to a round, you'll never pay a fee. Also when you are remixing if you cannot pay the full fee with your input, then you only pay as much as you have, but if the change amount leftover would be too small, then that's also added to the fee. Currently the minimum change amount to be payed out is 0.7% of the base denomination (~0.1BTC.)  

Reviewer: payed -> paid
```suggestion
There are also edge cases where you don't pay the full fee or where you pay more. For example if you're the smallest registrant to a round, you'll never pay a fee. Also when you are remixing if you cannot pay the full fee with your input, then you only pay as much as you have, but if the change amount leftover would be too small, then that's also added to the fee. Currently the minimum change amount to be paid out is 0.7% of the base denomination (~0.1BTC.)  
```

FAQ for [Wasabi Wallet](https://github.com/zkSNACKs/WalletWasabi)
By [6102bitcoin](https://twitter.com/6102bitcoin)

Note: Many of these Q&A have been copied from real users, see footer for acknowledgements. 

# Index
- Pre-Install
- Install
- Pre-Mix
- Mixing
- Post-Mix
- Meta
- Acknowledgements

# Pre-Install

### Who is behind Wasabi?

The Company that is developing Wasabi is zkSNACKs LTD ([twitter](@Zksnacks_LTD) | [website](https://zksnacks.com/))  
Members as of 2019.05.08:

| Name              | Twitter                                                 | Position			   |
| ------------------| --------------------------------------------------------|	-----------------------|
| Gergely Hajdú 	| [@gergely_hajdu](https://twitter.com/gergely_hajdu) 	  | Co-CEO & Co-Founder	   |
| Bálint Harmat		| [@bharmat84](https://twitter.com/bharmat84) 			  | Co-CEO & Co-Founder    |
| Adam Ficsór		| [@nopara73](https://twitter.com/nopara73)	              | Developer & Co-Founder |

> | Lucas Ontivero	| [@lontivero](https://twitter.com/)                      | Developer & Co-Founder |

Reviewer: @lontivero @bharmat @hgergoil It's ok if we call Lucas Co-Founder, right? He was here when we started the company.

Reviewer 2: This is true but legally not correct.

| Dan Walmsley   	| ([@danwalmsley](https://github.com/danwalmsley))        | Developer              |

> | Dávid Molnár   	| ([@molnardavid84](https://twitter.com/molnardavid84))   | Developer              |

Reviewer: The members section needs constant maintenance, maybe we should remove it.

Reviewer 2: Adam I think you are CTO not developer.

Author: It would require some maintenance. If you don't think it is worth it then lets remove it.

Author: Removed

# Install

### How do I install Wasabi?

Follow [this guide](https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Documentation/Guides/InstallInstructions.md)

> If you use Windows there is a [video guide](https://youtu.be/D8U53PFEsVk).

Reviewer: There are video guides for every OS, linked in the document above. I Think this line is unnecessary. (And needs maintenance.)
```suggestion
```

### Do I need to run Tor?

> All Wasabi network traffic goes via Tor by default - no need to set up Tor yourself. If you do already have Tor, and it is running, then Wasabi will try to use that first.  

Reviewer: ```suggestion
All Wasabi network traffic goes via Tor by default - no need to set up Tor yourself. If you do already have Tor, and it is running, then Wasabi will try to use that first.  
You can turn off Tor in the Settings. Note that in this case you are still private, except when you coinjoin and when you broacast a transaction. In the first case, the coordinator would learn the links between your inputs and outputs based on your IP address, in the second case, if you happen to broadcast a transaction of yours to a full node that is spying on you, it will learn the link between your transaction and your IP address.
```

# Pre-Mix

### My wallet can't send to Bech32 addresses - what wallets can I use instead? 

Wasabi generates Bech32 addresses only. These addresses start with the characters bc1... Some wallets/exchanges don't yet support this type of address and my give an error message (e.g. "unknown bitcoin address"). The solution is to manage your funds with a wallet which does support Bech32. Of [all the wallets](https://en.bitcoin.it/wiki/Bech32_adoption) that support Bech32, only the following wallets have Coin Control;

| Platform 						| Recommended 																					| Other										|
| ---------						| -----------------------------------------------------	|------------------------	|
| Mobile (Android)		| [Samourai](https://samouraiwallet.com/)								| [Electrum](https://play.google.com/store/apps/details?id=org.electrum.electrum)								|
| Desktop 						| [Bitcoin Core](https://bitcoincore.org/en/download/)	| [Electrum](https://electrum.org)	|

> I am not aware of any wallets for iOS which support coin control. If you are aware of any other Wallets with coin control please make a pull request.

Reviewer: While I see why you think coin control is important here, I think it's not that important here. Without going into the details, this section would need maintenance, so I'd vote for removing the end of it. The link to bech32 adoption however should be present.

Author: Removed.

> Be careful. If you send all your coins from an old wallet to a new wallet (from the table above) in one transaction then you will merge all your coins which is bad for privacy. Instead, **import your seed into one of the wallets in the table.**

Reviewer: This is, while good, an impractical advice unfortunately :(

Author: How so?

Reviewer: 1. 99% of users would fail to figure out what to do.
2. Gap limit issues.
3. Core is not compatible with other wallets.
4. Wasabi is only looking at bech32 keys, this can result in a lot of confusion.
5. Keypath issues.
6. xpub format issues: xpub hex, xpub, ypub, zpub, etc are not standards and used differently wallet by wallet.
7. Import Wasabi seed to a mobile wallet and you doxxed yourself and all your past mixes to the company.

And these are just from the top of my head.

# Mixing

### What are the fees?

> You pay currently pay a fee of 0.003% * Anonymity set. If the coin anonymity set of a coin is 50 then you paid 0.003% * 50 (=0.15%). If you set the target anonymity set to 53 then wasabi will continue mixing until this is reached, so you may end up with an anonymity set of say 60, and you will pay 0.003% * 60 (=0.18%).

Reviewer: ```suggestion
You currently pay a fee of 0.003% * anonymity set. If the anonymity set of a coin is 50 then you paid 0.003% * 50 (=0.15%). If you set the target anonymity set to 53 then wasabi will continue mixing until this is reached, so you may end up with an anonymity set of say 60, and you will pay 0.003% * 60 (=0.18%).  
There are also edge cases where you don't pay the full fee or where you pay more. For example if you're the smallest registrant to a round, you'll never pay a fee. Also when you are remixing if you cannot pay the full fee with your input, then you only pay as much as you have, but if the change amount leftover would be too small, then that's also added to the fee. Currently the minimum change amount to be payed out is 0.7% of the base denomination (~0.1BTC.)  
It is also possible that you get more back from mixing than you put in. This happens when network fees go down from when the round started and when the round ended. In this case, the difference is split between the active outputs of the mix.
```

### What is the Anonymity Set?

The anonymity set is effectively the size of the group you are hiding in. 

If 3 people take part in a CoinJoin (with equal size inputs) and there are 3 outputs then each of those output coins has an anonymity set of 3.

```
0.1 BTC (Alice)       0.1 BTC (Anon set 3)
0.3 BTC (Bob) 	  ->  0.1 BTC (Anon set 3)
0.4 BTC (Charlie)     0.1 BTC (Anon set 3)
                      0.2 BTC (Change Coin Bob)
                      0.3 BTC (Change Coin Charlie)
```

There is no way to know which of the anon set output coins are owned by which of the input owners.

> All an observer knows is that a specific anon set output coin is owned by one of the owners of one of the input Coins i.e. 3 people - hence an anonymity set of 3.

Reviewer: ```suggestion
All an observer knows is that a specific anon set output coin is owned by one of the owners of one of the input Coins i.e. 3 people - hence an anonymity set of 3.  
Your Wasabi software has limited information on what the anonymity set should be, so the anonymity set that the software presents you is just an estimation, not an accurate value. With Wasabi we are trying to do lower estimations, rather than upper ones.
```

### How Do I change the default number of mixing rounds (the Anonymity Set)?

> Go to File>Open>Config and change the value of **N** in "MixUntilAnonymitySet": **N** to the Anonymity Set you desire. More is better, (arguably only up to a point). Remember that you pay a fee proportional to the Anonymity Set.

Reviewer: ```suggestion
Go to `File`>`Open`>`Config File` and change the value of `PrivacyLevelSome`, `PrivacyLevelFine` or `PrivacyLevelStrong`. These are mapping to the privacy shields you see in the wallet. Then you can change your `Target` in the `CoinJoin` tab. Remember that you pay a fee proportional to the Anonymity Set.
```

"arguably up to a point" that's bullshit, I've seen people doing mental gymnastics on this, but makes absolutely no sense:)

Author: I'll merge the suggestion made by @MaxHillebrand in the version in my repo (https://github.com/6102bitcoin/FAQ/pull/2/commits/c77ec41860ee6b2a2ad7dfcf2e3222b3bfbfef31)

Similarly, with a 0.7 BTC input you would expect the following outputs: ~ 0.1, ~ 0.2, ~ 0.4 BTC. The possible values of equal output that can be created are 0.1 x 2^n where is a positive integer (or zero).

### Why is the minimum mixing amount a weird number?

The output value changes each round to ensure that you can que a coin and have it remix (mix over and over again - increasing the anonymity set, improving privacy). As a result the round mixing amount will often be a specific number which generally decreases as the rounds proceed, with a reset once a lower bound is reached. 

# Post-Mix

### What do I do now that I have mixed my coins?

Wasabi is a very good wallet, and it is advisable to manage your funds using the wallet once you have mixed. If you are going to send your funds to another wallet (say, a mobile wallet for convenience) then there are a couple of important things to consider.
- Unless the wallet has coin control you will merge your coins which damages privacy. 
- Unless the wallet connects to your own full node you will leak information to the server supplying the blocks/filters.
As a result, there are few wallets that are suitable. 

> Personally, I recommend sending your Zerolink Change to [Samourai Wallet](https://samouraiwallet.com/) or [Blockstream Green](https://blockstream.com/green/) for small-spends where you are less concerned about revealing any information. 

Reviewer: Can you remove personal recommendations? If you want to recommend then maybe recommend ones those can connect to your own full node, but those are fairly impractical, so removing the specific recommendations and leave the general recommendations may be the best.

Reviewer: You may want to also add a recommendations of trying to spend whole coins whenever possible.

### Am I safe to send my mixed coins to my hardware wallet?

Most hardware wallets communicate with servers to provide you with your balance. This reveals your public key to the server, which damages your privacy - the hardware company can now theoretically link together all your addresses. As a result **it is not recommended** that you send your mixed coins to an address associated with your hardware wallet unless you are confident that you have set up your hardware wallet in a way that it does not communicate with a 3rd party server (see below). 

> You can however manage your hardware wallet with the Wasabi interface.

Reviewer: ```suggestion
You can however manage your hardware wallet with the Wasabi interface. Alternatively you can use your hardware wallet with Electrum, which connects to your Bitcoin Core full node through Electrum Personal Server.
```

### How can I set up my hardware wallet with Wasabi properly?

You can currently use the following hardware wallets **with Wasabi directly**.
- Coldcard
- Trezor Model T

> - Ledger Nano S

Reviewer: This requires maintenance. I'd say remove it. I am sure when I implement Trezor One I won't update this document for example. Furtermore everyone can see the supported hardware wallets when the user clicks on the hardware wallet tab.

Author: Simplified

Alternately you can use [electrum personal server](https://github.com/chris-belcher/electrum-personal-server). 

### Will I have issues spending my mixed coins? 

> Not at the moment, if Wasabi and other CoinJoin tools are used by enough people it is likely that this will never be an issue. See this more [comprehensive answer](https://www.reddit.com/r/WasabiWallet/comments/bggy03/will_coinjoined_coins_be_blacklisted_in_the_future/ell04nn?utm_source=share&utm_medium=web2x). Reports of users having their account banned by Bitfinex are being [investigated](https://www.reddit.com/r/WasabiWallet/comments/bggy03/will_coinjoined_coins_be_blacklisted_in_the_future/elld52h?utm_source=share&utm_medium=web2x), though it has been [confirmed](https://old.reddit.com/r/WasabiWallet/comments/beqj8r/bitfinex_lock_account/el99fun/) that this is not a universal rule being applied and appears only have affected a [single user](https://old.reddit.com/r/WasabiWallet/comments/beqj8r/bitfinex_lock_account/).

Author: Removed Bitfinex stuff

> Not at the moment, if Wasabi and other CoinJoin tools are used by enough people it is likely that this will never be an issue. See this more [comprehensive answer](https://www.reddit.com/r/WasabiWallet/comments/bggy03/will_coinjoined_coins_be_blacklisted_in_the_future/ell04nn?utm_source=share&utm_medium=web2x). Reports of users having their account banned by Bitfinex are being [investigated](https://www.reddit.com/r/WasabiWallet/comments/bggy03/will_coinjoined_coins_be_blacklisted_in_the_future/elld52h?utm_source=share&utm_medium=web2x), though it has been [confirmed](https://old.reddit.com/r/WasabiWallet/comments/beqj8r/bitfinex_lock_account/el99fun/) that this is not a universal rule being applied and appears only have affected a [single user](https://old.reddit.com/r/WasabiWallet/comments/beqj8r/bitfinex_lock_account/).

Author: Removed

|	Exchange	|	Most recent Confirmed Accepted CoinJoin	|	Source																											|
|	--------	|	---------------------------------------	|	-------																											|
|	Coinbase	| 18/01/2019															|	[Link](https://twitter.com/a48/status/1086265253212639232) 	|

> **Note: This is for reference only, and by nature this is not evidence that you will be fine sending a mixed output to the exchange, only that someone else has at some time in the past. Also - don't use coinbase.**

Author: Removed

### What do I do with the Zerolink change?

There are no hard and fast rules for what to do with the change. It is important to note that the change may be linked to your identity and should be treated as a kind of toxic waste (handled with great care).

**Warning**

> Mixing change from multiple originating sources **will decrease your privacy** because it will link these transactions. This is true even if you mix it with a mixed coin. It is very important that you don't send different coins to the same receiving address, even if performed as separate transactions - as this will also link the coins damaging your privacy.

Author: Rephrased

- Mix with [Joinmarket](https://github.com/JoinMarket-Org/joinmarket-clientserver).
- Donate them (e.g. [to the EFF](https://www.eff.org/))
- Spend them on something that isn't a particular privacy risk (eg. gift cards).
- Open a lightning channel. 
- The ultimate solution is to 'close the loop' i.e. spend a change coin without merging it with other coins don't generate it in the first place by sending whole coins. 

# Meta

### Does Wasabi have a warrant canary?

The nature of Wasabi is that you shouldn't need to trust the devs or the wasabi coordinating server, as you can verify that the code does leak information to anyone. The dev's have gone to great lengths in an attempt to ensure that the coordinator can't steal funds or harvest information (for example, the outputs sent from your wasabi wallet are blinded, meaning that even the wasabi server can't link the outputs to the inputs). 

The only known possible 'malicious' actions that the server *could* perform are two sides of the same coin;
- blacklisted UTXO's
Though this would not affect the users who are able to successfully mix with other 'honest/real' peers. 
- Sybil Attack 

> The follow-up concern is the inverse of the above. It is possible that the server could *only* include one 'honest/real' coin in the mix and supply the other coins themselves. This would give a false sense of security, **but it would not worsen the existing privacy of the coin**. It has been argued that this 'attack' would be very costly in terms of fees because the number of coins being mixed is verifiable. Though it is true that fees would have to be paid to zkSNACKs every round this does not matter if it is zkSNACKs that is acting maliciously (as they get the funds back). Typical rounds currently have <100 people per mix, with the minimum input being ~0.1 BTC with a fee of 0.003% per anonymity set. Taking the 'worst case' (100 people, each mixing 0.1 BTC) gives 0.03 BTC per round. This is not prohibitive and is thus a valid concern. That said, if multiple chain-analysis companies attempt to flood the zkSNACKs mix (to decrease the true anonymity set) they will hinder each other's efforts (unless they are cooperating). 

Author: True. I have changed the title to Targeted Sybil attack and noted that it would be noticeable. I have also added that link.

# Errors

### 'Backend won't connect'

> All Wasabi network traffic goes via Tor. When Tor has issues Wasabi has issues. If the Tor Hidden Service directory does down (which is does occasionally) Wasabi now has a fall-back back to the coordinator server without a hidden service (but still over Tor). It is easiest to wait and try again some hours later. Alternatively you can go to the config file and change replace "MainNetBackendUriV3": "http://wasabiukrxmkdgve5kynjztuovbg43uxcbcxn6y2okcrsg7gb6jdmbad.onion/", with "MainNetBackendUriV3": "https://wasabiwallet.io/",

Author: Would it not be worth keeping the guidance to just wait a few hours first?

> All Wasabi network traffic goes via Tor. When Tor has issues Wasabi has issues. If the Tor Hidden Service directory does down (which is does occasionally) Wasabi now has a fall-back back to the coordinator server without a hidden service (but still over Tor). It is easiest to wait and try again some hours later. Alternatively you can go to the config file and change replace "MainNetBackendUriV3": "http://wasabiukrxmkdgve5kynjztuovbg43uxcbcxn6y2okcrsg7gb6jdmbad.onion/", with "MainNetBackendUriV3": "https://wasabiwallet.io/",

Author: Added

FAQ for [Wasabi Wallet](https://github.com/zkSNACKs/WalletWasabi)
By [6102bitcoin](https://twitter.com/6102bitcoin)

Note: Many of these Q&A have been copied from real users, see footer for acknowledgements. 

# Index
- Pre-Install
- Install
- Pre-Mix
- Mixing
- Post-Mix
- Meta
- Acknowledgements

# Pre-Install

### Who is behind Wasabi?

The Company that is developing Wasabi is zkSNACKs LTD ([twitter](@Zksnacks_LTD) | [website](https://zksnacks.com/))

The following people are listed on the company website:

| Name						| Twitter 						 																		| Position							|
| ----------------| --------------------------------------------------------|	----------------------|
| Hajdú Gergely 	|	[@gergely_hajdu](https://twitter.com/gergely_hajdu) 		| CEO & Co-Founder			|
| Harmat Bálint		|	[@bharmat84](https://twitter.com/bharmat84) 						| CEO & Co-Founder			|
| Adam Ficsor			|	[@nopara73](https://twitter.com/nopara73)								| CTO										|
| Lucas Ontivero	| [@lontivero](https://twitter.com/)											| Lead Engineer					|

Other people have contributed to the code:
- David Molnar ([@molnardavid84](https://twitter.com/molnardavid84))

> - Dan Walmsley ([@danwalmsley](https://github.com/danwalmsley))

Reviewer: Well, the listing there is just outdated, but oh well, not sure I want to nitpick this, but I will anyway:)

Instead of making recommendation here, I'll just resolve this section myself.
